scrapyd/webservice.py

```diff
@@ -82,7 +83,7 @@ def render_POST(self, txrequest):
         version = txrequest.args['version'][0]
         eggf = StringIO(txrequest.args['egg'][0])
         self.root.eggstorage.put(eggf, project, version)
-        spiders = get_spider_list(project)
+        spiders = get_spider_list(project, version=version)
         self.root.update_projects()
         UtilsCache.invalid_cache(project)
```

Comment:
This is a bit too broad. It could potentially invalidate only 1 or 2 cache entries (instead of every entry for this project) but I don't think that performance justifies the extra code complexity.